gen3-integration-tests/services/graph.py

```diff
@@ -438,18 +437,109 @@ def query_record_with_path_to(self, from_node: dict, to_node: dict) -> str:
+            expected_status: expected api response status
+            invalid_authorization: If True set the authorization to invalid
+        """
+        min_sem_ver = "3.2.0"
+        min_monthly_release = "2023.04.0"
+        monthly_release_cutoff = "2020"
+
+        auth = Gen3Auth(refresh_token=pytest.api_keys[user], endpoint=pytest.root_url)
+        response = auth.curl(path=self.GRAPHQL_VERSION_ENDPOINT)
+        peregrine_version = response.json()["version"]
+        url = f"{pytest.root_url}/api/search/coremetadata/"
+
+        if peregrine_version:
+            try:
+                if Version(peregrine_version) < Version(min_sem_ver) or (
+                    Version(peregrine_version) >= Version(monthly_release_cutoff)
+                    and Version(peregrine_version) < Version(min_monthly_release)
+                ):
+                    # Legacy endpoint
+                    url = "{pytest.root_url}/coremetadata/"
+            except:
+                logger.error(
+                    "Can't parse or compare the peregrine version: don't user legacy url"
+                )
+
+        authorization = f"bearer {auth.get_access_token()}"
+        if invalid_authorization:
+            authorization = "invalid"
+        headers = {
+            "Authorization": authorization,
+            "Accept": format,
+        }
+        response = requests.get(url=url + file.indexd_guid, headers=headers)
+        assert response.status_code == expected_status, f"{response}"
+        return response
+
+    def verify_metadata_bibtex_contents(self, record, metadata):
+        """
+        Verifies the metadata bibtex contents is as expected
+        Args:
+            record: Graph Record
+            metadata: coremetadata information
+        """
+        metadata = metadata.content.decode()
+        assert (
+            record.props["file_name"] in metadata
+        ), f"file_name not matched/found.\n{record}\n{metadata}"
+        assert (
+            record.indexd_guid in metadata
+        ), f"object_id not matched/found.\n{record}\n{metadata}"
+        assert (
+            record.props["type"] in metadata
+        ), f"type not matched/found.\n{record}\n{metadata}"
+        assert (
+            record.props["data_format"] in metadata
+        ), f"data_format not matched/found.\n{record}\n{metadata}"
+
+    def see_metadata_error(self, metadata, message):
+        """
+        Verifies the metadata errors
+        Args:
+            metadata: coremetadata information
+            message: expected message
+        """
+        if "message" not in metadata.json().keys():
+            logger.error(f"Message key missing.\n{metadata.json()}")
+            raise
+        logger.info(metadata.json()["message"])
+        if message != metadata.json()["message"]:
+            logger.error(f"Expected message not found.\n{metadata.json()}")
+            raise
+
+    def verify_metadata_json_contents(self, record, metadata):
```

Comment:
similar comment about vague function names. this function only checks some specific fields, not all, so it's not generic. if we don't have a better term and we have to use the term "core metadata" to know what the functions/tests are doing, then let's do that